### PR TITLE
added semantic commit check and release action (#700)

### LINKED ISSUE #700: use github actions to create changelog and release the plugin

### PR DESCRIPTION
**Pull request by adietish**
fixes #700

### CODE REVIEW

**Comment by adietish** [comment]
@sbouchet copied those 2 scripts over from intellij-openshift-connector. Missing something?

**Comment by sbouchet** [comment]
> @sbouchet copied those 2 scripts over from intellij-openshift-connector. Missing something?

be sure that the gradle properties for versionning and channel are the same on the script and yes, this should be good

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+          git config user.name "GitHub Action"
+          if git diff --quiet; then
+            echo "No changes to commit."
+          else
+            git commit -sam "chore(skip-release): set version to $PLUGIN_VERSION"
+          fi
+          git tag $PLUGIN_VERSION
+          git push origin $PLUGIN_VERSION
+
+      # Set next SNAPSHOT version
```
Comment (sbouchet) [inline .github/workflows/release.yml:81]: uncomment those lines if your current IC looks good !

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      #    IFS="-" read -ra VERSION_PARTS <<< "$CURRENT_VERSION"
+      #    IFS="." read -ra VERSION_NUM <<< "${VERSION_PARTS[0]}"
+      #    ((VERSION_NUM[2]++))
+      #    NEW_VERSION="${VERSION_NUM[0]}.${VERSION_NUM[1]}.${VERSION_NUM[2]}-SNAPSHOT"
+      #    awk -v new_version="$NEW_VERSION" '/projectVersion=/{sub(/=.*/, "=" new_version)}1' gradle.properties > tmpfile && mv tmpfile gradle.properties
+      #    echo "Set $NEW_VERSION in gradle.properties"
+      #    git commit -sam "chore(skip-release): set version to $NEW_VERSION"
+      #    git push origin main
+
+      - name: Simple conventional changelog
```
Comment (sbouchet) [inline .github/workflows/release.yml:97]: here  we're using conventional changelog that you're not using to generate changelog. this must be changed to what you're currenlty doing, OR must use a check to be sure every commit as a correct changelog titile

Comment (adietish) [reply .github/workflows/release.yml:97]: @sbouchet: I added the check: `conventionalCheck.yml`. Maybe I should file this in a separate PR that we merge first?

Comment (sbouchet) [reply .github/workflows/release.yml:96]: yes, that makes sense to do this way.

Comment (adietish) [reply .github/workflows/release.yml:96]: @sbouchet: added the semantic commit check in a separate commit for this PR

**Review by sbouchet — APPROVED** [review]
LGTM

**Pull request merged**